boards/riscv/tlsr9518adk80d/tlsr9518adk80d.dts

```diff
@@ -0,0 +1,161 @@
+/*
+ * Copyright (c) 2021 Telink Semiconductor
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/dts-v1/;
+
+#include <telink_b91.dtsi>
+
+/ {
+	model = "telink,b91";
+	compatible = "telink,b91";
+
+	aliases {
+		led0 = &led_blue;
+		led1 = &led_green;
+		sw0 = &key_1;
+		pwm-led0 = &pwm_led_blue;
+		pwm-0 = &pwm0;
+	};
+
+	leds {
+		compatible = "gpio-leds";
+
+		led_blue: led_0 {
+			gpios = <&gpiob 4 GPIO_ACTIVE_HIGH>;
+			label = "LED Blue";
+		};
+
+		led_green: led_1 {
+			gpios = <&gpiob 5 GPIO_ACTIVE_HIGH>;
+			label = "LED Green";
+		};
+
+		led_white: led_2 {
+			gpios = <&gpiob 6 GPIO_ACTIVE_HIGH>;
+			label = "LED White";
+		};
+
+		led_red: led_3 {
+			gpios = <&gpiob 7 GPIO_ACTIVE_HIGH>;
+			label = "LED Red";
+		};
+	};
+
+	pwm_leds {
+		compatible = "pwm-leds";
+
+		pwm_led_blue: pwm_led_0 {
+			pwms = <&pwm0 0 0>;
+			label = "PWM LED Blue";
+		};
+	};
+
+	keys {
+		compatible = "gpio-keys";
+		key_1: button_1 {
+			label = "User KEY1";
+			gpios = <&gpioc 2 GPIO_PULL_DOWN>;
+		};
+	};
+
+	chosen {
+		zephyr,console = &uart0;
+		zephyr,shell-uart = &uart0;
+		zephyr,sram = &ram_dlm;
+		zephyr,flash = &flash;
+		zephyr,entropy = &trng0;
+		zephyr,flash-controller = &flash_mspi;
+	};
+};
+
+&cpu0 {
+	clock-frequency = <48000000>;
+};
+
+&ram_ilm {
+	reg = <0x00000000 0x00020000>;
+};
+
+&ram_dlm {
+	reg = <0x00080000 0x00020000>;
+};
+
+&flash {
+	reg = <0x20000000 0x100000>;
+};
+
+&gpiob {
+	interrupts = <26 1>;
```

Comment:
Are the interrupt numbers actually board specific, that seems odd?

Reply:
Yes, only IRQs 25, 26 and 27 could be assigned for GPIOs.